### PR TITLE
Add Unit Tests for Planners and GBP Update

### PR DESCRIPTION
**Pull request by jcnorby**
Global body planner changes:

-    Switch leap parameterization to SLIP-like profile
-    Add coarse leap refinement to promote better leaps
-    Add more detailed unit tests for operations on states and actions
-    Refactor ROS interface to handle new planning requests smoothly
-    Implement GlobalBodyPlan class to contain plan data and useful functions

### CODE REVIEW

File: global_body_planner/include/global_body_planner/global_body_plan.h [hunk 1]
```diff
@@ -20,63 +20,157 @@ class GlobalBodyPlan {
...
+   * @param[in] plan_status Status of this particular plan
+   * @param[in] start_state State at which this plan begins
+   * @param[in] dist_to_goal Distance to goal from plan end
+   * @param[in] state_sequence Discrete state sequence to be interpolated
+   * @param[in] action_sequence Discrete action sequence to be interpolated
+   * @param[in] dt Timestep for interpolation
+   * @param[in] t0 Start time of this plan section
+   * @param[in] planner_config Planning configuration parameters
+   */
+  void load(int plan_status, FullState& start_state, double dist_to_goal,
```
Comment (ardalantj) [inline global_body_planner/include/global_body_planner/global_body_plan.h:150]: loadPlanData?

File: global_body_planner/include/global_body_planner/global_body_plan.h [hunk 2]
```diff
@@ -20,63 +20,157 @@ class GlobalBodyPlan {
...
+    int primitive_id, quad_msgs::RobotPlan& msg);
 
-  void toMsg(quad_msgs::RobotPlan &robot_plan_msg,
-             quad_msgs::RobotPlan &discrete_robot_plan_msg);
+  /**
+   * @brief Load the entire plan into a plan message
+   * @param[out] robot_plan_msg Interpolated robot plan message
+   * @param[out] discrete_robot_plan_msg Discrete robot plan message
+   */
+  void toMsg(quad_msgs::RobotPlan& robot_plan_msg,
```
Comment (ardalantj) [inline global_body_planner/include/global_body_planner/global_body_plan.h:172]: robotPlanToMsg?

File: global_footstep_planner/src/global_footstep_planner.cpp [hunk 3]
```diff
@@ -281,7 +281,7 @@ void GlobalFootstepPlanner::updateDiscretePlan() {
 
   // publishDiscretePlan();
   // publishContinuousPlan();
-  // timer.report();
+  // timer.reportStatistics();
```
Comment (ardalantj) [inline global_footstep_planner/src/global_footstep_planner.cpp:284]: Ok with leaving useful debugging messages commented, just wanted to make sure you want it here.

**Comment by ardalantj** [comment]
@jcnorby Looks good to me. Just a few minor comments.

File: global_footstep_planner/src/global_footstep_planner.cpp (hunk 3 above)
Comment (jcnorby) [reply global_footstep_planner/src/global_footstep_planner.cpp:284]: Actually this entire package is pretty outdated, it was before local planner existed and was only there to add footsteps to a global plan for visualization/mpc. This seems like a decent PR to just remove it entirely.

File: global_body_planner/include/global_body_planner/global_body_plan.h [hunk 4]
```diff
@@ -20,63 +20,157 @@ class GlobalBodyPlan {
...
+   * @return Plan emptiness
+   */
+  inline bool isEmpty() const { return t_plan_.empty(); };
 
-  inline bool isWorseThan(int other_plan_status, double other_plan_length) {
+  /**
+   * @brief Check if this worse than another plan (less solved or longer)
+   * @return Relative plan quality
+   */
+  inline bool isWorseThan(int other_plan_status, double other_plan_length) const {
```
[the remark this answers is not in the document]
Comment (jcnorby) [reply global_body_planner/include/global_body_planner/global_body_plan.h]: This function didn't end up getting used so it has been removed.

File: global_body_planner/include/global_body_planner/global_body_plan.h (hunk 1 above)
Comment (jcnorby) [reply global_body_planner/include/global_body_planner/global_body_plan.h]: Fixed.

File: global_body_planner/include/global_body_planner/global_body_plan.h (hunk 2 above)
Comment (jcnorby) [reply global_body_planner/include/global_body_planner/global_body_plan.h]: Fixed.

**Review by ardalantj — APPROVED** [review]

**Pull request merged**